fastlane_core/lib/fastlane_core/build_watcher.rb

```diff
@@ -10,14 +10,17 @@ def wait_for_build_processing_to_be_complete(app_id: nil, platform: nil, train_v
         # Warn about train_version being removed in the future
         if train_version
           UI.deprecated(":train_version is no longer a used argument on FastlaneCore::BuildWatcher. Please use :app_version instead.")
+          app_version = train_version
         end
-        app_version = train_version
 
         # Warn about strict_build_watch being removed in the future
         if strict_build_watch
           UI.deprecated(":strict_build_watch is no longer a used argument on FastlaneCore::BuildWatcher.")
         end
 
+        platform = Spaceship::ConnectAPI::Platform.map(platform) if platform
+        UI.message("Waiting for processing on... app_id: #{app_id}, app_version: #{app_version}, build_version: #{build_version}, platform: #{platform}")
```

Comment:
If I understood line 21 correctly, this could be missing `platform` if `platform` isn't defined before already. Might that be a problem?

Reply:
This _could_ potentially? 🤔 We weren't explicitly erroring out on no platform before. Not sure if we should be now but this message and the error message of multiple builds _should_ be a good indication that `platform` is needed **if** someone is calling this manually (outside of `pilot`)

Reply:
ah yeah, we were using the same code to set and use `platform` before - so this should not break anything.